src/core/futures.rs

```diff
@@ -23,7 +23,10 @@ macro_rules! fry {
     ($res:expr) => {
         match $res {
             Ok(elt) => elt,
-            Err(e) => return futures::done(Err(From::from(e))),
+            Err(e) => {
+                use $crate::core::futures::FutureExt;
+                return futures::done(Err(From::from(e))).into_box()
```

Comment:
True. But without `into_box`, it's unusable in cases like this:

```rust
fut.and_then(move |a| {
    let b = fry!(process(a));
    something_that_returns_future(b)  
})
```
Besides, in the example you gave, you can replace `fry!` with `try!` and it would still work the same.